docs/design/rubygems.md

```diff
@@ -0,0 +1,524 @@
+
+##### Other configuration that was considered
+
+- `BUNDLE_ALLOW_OFFLINE_INSTALL` is not working either with `bundle install` for some reason, which could be probably
+the most logical solution in this case.
+
+### Generating the SBOM
+
+#### Main package metadata
+
+Ruby uses [Gem::Specification](https://guides.rubygems.org/specification-reference/) as a means of defining a Gem's
+metadata, and it is usually defined in a `{gem-name}.gemspec` file. This file is not mandatory, though, and when it
+exists, it needs to be explicitly imported in the `Gemfile`:
+
+```
+source "https://rubygems.org"
+
+gemspec
+```
+
+When the `.gemspec` file exists and is properly imported, it will be listed in the `Gemfile.lock` as a `PATH`
+dependency:
+
+```
+PATH
+  remote: .
+  specs:
+    tmp (0.1.2)
+```
+
+Since the `remote` will always point to `.` in case of the main package [^main-package], we can safely use it to get
+the `name` and `version` for the SBOM component. In case this block is absent, we will need to fallback to the
+repository's remote `origin` to retrieve the main package's name, and leave the version empty, since it is not a
+mandatory field.
+
+[^main-package]: In Cachi2's terms, the **main package** is the path in the repository that is currently being
+  processed.
+
+#### PURLs
+
+Also check the Ruby PURL [specification](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#gem).
+
+##### Standard Gem
+```txt
+pkg:gem/my-gem-name@0.1.1
+```
+
+##### Git dependency
+
+```txt
+pkg:gem/my-git-dependency?vcs_url=git%2Bhttps://github.com/my-org/mygem.git%26487618a68443e94d623bb585cb464b07d36702
+```
+
+The metadata for a Git dependency can be read from the `Gemfile.lock`:
+
+```
+GIT
+  remote: https://github.com/my-org/mygem.git
+  revision: 26487618a68443e94d623bb585cb464b07d36702
+  specs:
+    json-schema (3.0.0)
+      addressable (>= 2.4)
+```
+
+##### Path dependency
+
+```txt
+pkg:gem/my-path-dependency?vcs_url=git%2Bhttps://github.com/my-org/mygem.git%40b6f47bd07e669c8d2eced8015c4bfb06db49949#subpath
+```
+
+The PURL is formed by the current repository remote origin URL and ref, and the subpath that is specified in the
+`Gemfile.lock`:
+
+```ruby
+PATH
+  remote: subpath
+  specs:
+    my-path-dependency (1.0.0)
+```
+
+### Summary
+
+- define models for RubyGems as the new package manager
+- design high-level code structure into multiple modules
+- parse all gems from `Gemfile.lock`
+- implement metadata parsing either from Git origin url or `Gemfile.lock`
+- download all gems from rubygems.org, including Bundler
+- download all gems from Git repositories
+- validate path dependencies are relative to the project root
+- handle Bundler configuration options and environment variables
+- generate PURLs for all dependencies
+- add integration and e2e tests
+- add documentation
+- implement checksum parsing and validation when prefetching (follow-up)
```

Comment:
Can we add all other follow-up tasks here? Ideally, separately for better readability.

Reply:
Done